fuel/datasets/hdf5.py

```diff
@@ -411,21 +616,27 @@ def _in_memory_get_data(self, state=None, request=None):
         return data, shapes
 
     def _out_of_memory_get_data(self, state=None, request=None):
+        if not isinstance(request, (slice, list)):
```

Comment:
Although true that they aren't safe, I do believe that they are significantly faster for some HDF5 implementations. My suggestion was to re-implement support for slices, but perhaps make it optional through a flag. That way, new users won't get bitten by it, but power users have the option to speed things up easily.